### PR TITLE
Feat: support complicated addon parameter

### PR DESCRIPTION
**Pull request by wangyikewxgm**
Addon parameter support  complicated parameter such as [struct, array].

eg:

**list parameter**: vela addon enable xxx imagePullSecrets={a,b,c}.  
**struct parameter**: vela addon enable xxx image.Repo=www.test.com image.tag=1.1 the args will translate to 

```json
{
  image: {
    repo: www.test.com
    tag: 1.1
  }
}
```

Signed-off-by: 楚岳 <wangyike.wyk@alibaba-inc.com>

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -69,19 +69,19 @@ require (
 	gopkg.in/src-d/go-git.v4 v4.13.1
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
 	gotest.tools v2.2.0+incompatible
-	helm.sh/helm/v3 v3.6.1
+	helm.sh/helm/v3 v3.7.2
```
Comment (barnettZQG) [inline go.mod:72]: Is it necessary to upgrade these packages?

**Comment by zzxwill** [comment]
Which addon will need this feature?

**Comment by zzxwill** [comment]
> Which addon will need this feature?

Got the answer from https://github.com/oam-dev/catalog/pull/243.

**Review by wonderflow — APPROVED** [review]
LGTM， please fix CI

File: go.mod (hunk 1 above)
Comment (wangyikewxgm) [reply go.mod:72]: Yes, version 3.6.1  package doesn't have this function.

Comment (barnettZQG) [reply go.mod:72]: the ParseIntoString function?  it exists in 3.6.1. `vela install` also depends on this function.

Comment (wangyikewxgm) [reply go.mod]: find it thanks。

**Review by zzxwill — APPROVED** [review]

**Comment by StevenLeiZhang** [comment]
Hi @wangyikewxgm , I tried this new feature, but hit one issue.
1. input list arguments for 'vela addon enable'
```
vela addon enable velaux imagePullSecrets={myreg}
```
2.The list is stored in Secret
 `"imagePullSecrets": "[myreg]",`

3.curl api-server
`curl 127.0.0.1:8000/api/v1/addons/velaux/status`
this api can not parse imagePullSecrets to List. It is string in 
```
 "args": {
  "imagePullSecrets": "[myreg]",
  "repo": "hub.jdcloud.com",
  "serviceType": "NodePort",
  "version": "v1.2.3"
 },

```

4.if you open velaux addon's details page on VelaUX webpage
The webpage will crash.

**Comment by wangyikewxgm** [comment]
> Hi @wangyikewxgm , I tried this new feature, but hit one issue.
> 
> 1. input list arguments for 'vela addon enable'
> 
> ```
> vela addon enable velaux imagePullSecrets={myreg}
> ```
> 
> 2.The list is stored in Secret `"imagePullSecrets": "[myreg]",`
> 
> 3.curl api-server `curl 127.0.0.1:8000/api/v1/addons/velaux/status` this api can not parse imagePullSecrets to List. It is string in
> 
> ```
>  "args": {
>   "imagePullSecrets": "[myreg]",
>   "repo": "hub.jdcloud.com",
>   "serviceType": "NodePort",
>   "version": "v1.2.3"
>  },
> ```
> 
> 4.if you open velaux addon's details page on VelaUX webpage The webpage will crash.

Thanks for report, What's your vela's version. I have try in 1.2.3 but cannot regenerate this bug. 

And this way storage list parameter is fine. Add list parameter in webpage just like this.

And I have submit a new issue  in velaux to track this: https://github.com/oam-dev/velaux/issues/372

**Pull request merged**